### ISSUE TITLE
Flutter Doctor windows component should describe which components were missing

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#55183]
Going through the setup on a new PC for windows development, I set up the visual studio dependencies - except for the incorrect version of the windows SDK. currently the `flutter doctor` text is the same for partial or full installation, so I had to read through the component list very carefully.

```
[!] Visual Studio - develop for Windows (Visual Studio Community 2019 16.5.4)
    X Visual Studio is missing necessary components. Please re-run the Visual Studio installer for the "Desktop
      development with C++" workload, and include these components:
        MSBuild
        MSVC v142 - VS 2019 C++ x64/x86 build tools
         - If there are multiple build tool versions available, install the latest
        Windows 10 SDK (10.0.17763.0)
```
Two possible fixes here:

1. Break out each component into its own line item. Then I can skip checking for msbuild and and MSVC

2. Introduce the concept of a partial match for a component. Essentially "Hey you installed the wrong version of the windows SDK" instead of "I couldn't find the specific windows SDK version".

**Comment by stuartmorgan** [comment]
> Introduce the concept of a partial match for a component. Essentially "Hey you installed the wrong version of the windows SDK" instead of "I couldn't find the specific windows SDK version".

See https://github.com/flutter/flutter/issues/50487; if we know they have some version of the Windows 10 SDK, then we're already done.

Once we do that, the instructions can probably just become "install the desktop development workload", since I think all those components are default selected for the workload.

> Break out each component into its own line item. Then I can skip checking for msbuild and and MSVC

The reason I haven't already done that is that I'm not happy with any approach I can find for doing it. The options I know of are:
1. Install [a third-party utility](https://github.com/microsoft/vswhere/issues/199#issuecomment-531875699) before we can even run doctor.
1. Discard all the `vswhere` component checks and figure out how to manually check for everything we need (basically #50487, but for every current and future component we ever require, which is going to be messy and fragile).
1. Instead of running one `vswhere` query with all the components, run N queries, each with one component and without the `--latest` flag, and then do set analysis on all the results to figure out if any VS installation has everything, and if not what is missing.

**Comment by stuartmorgan** [comment]
The instructions are now just to install the desktop workload with default items.

Leaving open since if VS is detected but missing something we do still list the workload and some components, so there would be some value to this, but it should come up far less often now.

**Comment by stuartmorgan** [comment]
Turns out I had filed this as https://github.com/flutter/flutter/issues/44193 a while ago.
